packages/webapp/src/components/cards/baseCard.tsx

```diff
@@ -0,0 +1,69 @@
+import React from "react"
+import { CardPlacement } from "src/store/types"
+import { useSortable } from "@dnd-kit/sortable"
+import DraggedCard from "./draggedCard"
+import BoardCard from "./boardCard"
+import HandCard from "./handCard"
+import { CSS } from "@dnd-kit/utilities"
+
+interface BaseCardProps {
+  id: number
+  className?: string
+  handHovered?: boolean
+  placement: CardPlacement
+}
+
+const BaseCard: React.FC<BaseCardProps> = ({
+  id,
+  handHovered,
+  placement,
+}) => {
+  const {
+    attributes,
+    listeners,
+    setNodeRef,
+    isDragging,
+    transform,
+    transition,
+  } = useSortable({
+    id: placement === CardPlacement.BOARD ? `B-${id}` : `H-${id}`,
+  })
+
+  const sortableStyle = {
+    transform: CSS.Transform.toString(transform),
+    transition,
+  }
+
+  const renderCardContent = () => {
+    switch (placement) {
+      case CardPlacement.HAND:
+        return (
+          <HandCard
+            id={id as number}
+            handHovered={handHovered}
+            isDragging={isDragging}
+            ref={setNodeRef}
+          />
+        )
+      case CardPlacement.BOARD:
+        return <BoardCard id={id} ref={setNodeRef} />
+      case CardPlacement.DRAGGED:
+        return <DraggedCard id={id} ref={setNodeRef} />
+      default:
+        return null
+    }
+  }
+  return (
+    <div
+      className={`${"shadow-2xl z-[50] flex  max-w-[24rem] cursor-pointer flex-col items-center justify-evenly rounded-lg bg-gray-900 transform translateY-[-50%]"}`}
+      style={sortableStyle}
+      ref={setNodeRef}
```

Comment:
The only parameter that tends to be null here is `id` and that's if it's a `bignumber` out the `number` type's safe range (current placeholder logic for the JSON list. 
The `placement` parameter can be one of 3 states, and is assigned a ref only if it's in `BOARD` or `HAND` (a simple placeholder is rendered if it's being `DRAGGED`).

Reply:
Mmh, I'm not sure this completely answers the question.

Is this assumption right?

> I'm guessing setNodeRef is passed to both the div and the card so that if there is no card, then the ref will be set to the outer div.

If it is, does this means we could remove `setNodeRef` from the `div`, since in principle we will never pass an `id` that exceeds the number safe range currently?